src/main/java/su/litvak/chromecast/api/v2/Channel.java

```diff
@@ -159,8 +164,20 @@ public void run() {
                                         LOG.warn("Unable to process request ID = {}, data: {}", requestId, jsonMSG);
                                     }
                                 }
-                            } else if (parsed.has("responseType") && parsed.get("responseType").asText().equals("PING")) {
+                            } else if (message.getNamespace().equals("urn:x-cast:com.google.cast.tp.heartbeat")
+                                    && parsed.has("responseType") && parsed.get("responseType").asText().equals("PING")) {
                                 write("urn:x-cast:com.google.cast.tp.heartbeat", StandardMessage.pong(), DEFAULT_RECEIVER_ID);
+                            } else if (message.getNamespace().equals("urn:x-cast:com.google.cast.tp.connection")
+                                    && parsed.has("responseType") && parsed.get("responseType").asText().equals("CLOSE")) {
+                                // Determine whether the close event was requested by this sender
+                                ((ObjectNode) parsed).put("requestedBySender", stopRequested);
```

Comment:
No. IIRC, it receives this CLOSE event when the Backdrop starts. There is no senderId in that moment though.